### PR TITLE
[Eth flow]#1290 disable for sc wallets

### LINKED ISSUE #1290: [ETH] Disable Eth Flow for Smart Contract Wallets
Related to #1289

Disable Limit Orders for Smart Contract Wallets.

It should default to the previous flow. Smart contract wallets can bundle transactions. We can still reiterate and add the flow if we see its important in the future.

### PR DESCRIPTION
**Pull request by alfetopito**
# Summary

Closes #1290 
Depends on https://github.com/cowprotocol/cowswap/pull/1367

Disabling Eth flow for Smart contract wallets

  # To Test

1. Enable the feature flag with `localStorage.setItem('enableEthFlow', '1')`
2. Sell ETH against another token
* Eth flow should be displayed
Screen Shot 2022-11-10 at 14 55 36
3. Connect an EOA
* Eth flow should still be available
4. Connect a smart contract wallet (Gnosis Safe for example)
* The previous wrap flow should be displayed
Screen Shot 2022-11-10 at 14 55 50

### CODE REVIEW

File: src/cow-react/common/hooks/useIsSmartContractWallet.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ function useCheckIsSmartContract(): boolean | undefined {
   return data
 }
 
-export default function useIsSmartContractWallet(): boolean {
+export function useIsSmartContractWallet(): boolean {
```
Comment (alfetopito) [inline src/cow-react/common/hooks/useIsSmartContractWallet.ts:26]: We'll get rid of default exports (eventually)

Comment (alfetopito) [reply src/cow-react/common/hooks/useIsSmartContractWallet.ts:26]: Also moved the file to our own part

**Review by anxolin — COMMENTED** [review]
The PR works like a charm!

However, I think the base branch is incorrect. This PR includes the ABIs, the generation of typechain, and many other unrelated things

**Comment by alfetopito** [comment]
> The PR works like a charm!
> 
> However, I think the base branch is incorrect. This PR includes the ABIs, the generation of typechain, and many other unrelated things

Rebased against develop

**Comment by elena-zh** [comment]
Hey @alfetopito , works great!
However, I noticed that slippage value is not changed when I switch from 
 - EOA: ETH-ERC20 token (remains 2%)
 - EOA ETH to SC ETH flow
Is this OK for the current PR?

**Comment by alfetopito** [comment]
> Hey @alfetopito , works great! However, I noticed that slippage value is not changed when I switch from
> 
>     * EOA: ETH-ERC20 token (remains 2%) **UPD**: I reported [ETH flow:  the value in the deadline/slippage is not saved when reload the page #1379](https://github.com/cowprotocol/cowswap/issues/1379) issue for this
> 
>     * EOA ETH to SC ETH flow (**steps**: connect to MM, select ETH, change wallet to SC using WC)
> 
> 
> Is this OK for the current PR?
No, is should not happen.
Will take a look

**Review by shoom3301 — APPROVED** [review]

**Comment by alfetopito** [comment]
> Hey @alfetopito , works great! However, I noticed that slippage value is not changed when I switch from
> 
>     * EOA: ETH-ERC20 token (remains 2%) **UPD**: I reported [ETH flow:  the value in the deadline/slippage is not saved when reload the page #1379](https://github.com/cowprotocol/cowswap/issues/1379) issue for this
> 
>     * EOA ETH to SC ETH flow (**steps**: connect to MM, select ETH, change wallet to SC using WC)
> 
> 
> Is this OK for the current PR?

@elena-zh I've tried now and the second case is not reproducible to me.

As for the first case (#1379) I'll address in another PR as it's not related to SC but to the deadline and slippage stuff.

**Comment by elena-zh** [comment]
Hey @alfetopito , 

> @elena-zh I've tried now and the second case is not reproducible to me.

here is the video with the steps I do in order to reproduce this issue: https://www.loom.com/share/5cd0debe387347569bc2f52273ab8616

**Comment by alfetopito** [comment]
> Hey @alfetopito ,
> 
> > @elena-zh I've tried now and the second case is not reproducible to me.
> 
> here is the video with the steps I do in order to reproduce this issue: https://www.loom.com/share/5cd0debe387347569bc2f52273ab8616

I've seen this a few times but I cannot consistently reproduce it :/

I tried a couple of times and it is fine most of them

https://user-images.githubusercontent.com/43217/202469838-dda55d6a-6b9a-4207-b9bc-413316634b93.mov

Is it any different if you first set the limit orders on, refresh the page, then connect the SC wallet?

**Comment by elena-zh** [comment]
> Is it any different if you first set the limit orders on, refresh the page, then connect the SC wallet?

So I set 'ETH flow' ON
Refreshed the page
Changed wallet to Gnosis Safe: the outcome is the same. 
Then I set the slippage to the default value, refreshed the page
Switched to EOA wallet and back --> 2%

See the video with all possible cases: https://www.loom.com/share/eab50e4d72db4ff2981c43ec37b89ec2

**Comment by alfetopito** [comment]
> > Is it any different if you first set the limit orders on, refresh the page, then connect the SC wallet?
> 
> So I set 'ETH flow' ON Refreshed the page Changed wallet to Gnosis Safe: the outcome is the same. Then I set the slippage to the default value, refreshed the page Switched to EOA wallet and back --> 2%
> 
> See the video with all possible cases: https://www.loom.com/share/eab50e4d72db4ff2981c43ec37b89ec2

Oooooooh, you are talking about the slippage?
Yes, that's wrong for sure!

I thought you were talking about the WRAP button showing up for the Safe.
That's what I can't reproduce and should NOT happen!

The ethflow slippage is not being saved, yes 😆 
I'll address that in a different PR

**Review by elena-zh — APPROVED** [review]
Approved then!

**Pull request merged**